### PR TITLE
Layout_intense_saving_in_year

### PR DESCRIPTION
**Pull request by hieplt-0914**
==> Tittle: Layout_intense_saving_in_year

==> Redmine:  https://edu-redmine.sun-asterisk.vn/issues/15696

==> Attachment File:Screenshot from 2019-07-30 15-15-12
Screenshot from 2019-07-30 15-15-06
Screenshot from 2019-07-30 15-15-01

### CODE REVIEW

**Comment by hieplt-0914** [comment]
ready

File: app/services/year_service.rb [hunk 1]
```diff
@@ -19,6 +19,9 @@ def perform_percent_expense
     cal_percent Settings.status_expense
   end
 
+  def saving_in_year
+    saving_year
+  end
   private
```
Comment (phucmh-1356) [inline app/services/year_service.rb]: trước private cách 1 dòng trắng

File: app/services/year_service.rb [hunk 2]
```diff
@@ -30,4 +33,14 @@ def cal_percent status
     AND NOW())").where(user_id: @user_id).where(status: status)
     @percent.to_json
   end
+
+  def convert_arr_to_fixnum num
+    num.inject{|n, d| n * 10 + d}
+  end
+
+  def saving_year
+    saving_id = User.by_user_id(@user_id).pluck(:saving_id)
```
Comment (phucmh-1356) [inline app/services/year_service.rb]: xóa ()

File: app/services/year_service.rb [hunk 3]
```diff
@@ -30,4 +33,14 @@ def cal_percent status
...
+    saving_in_year = Saving.by_saving_id(convert_arr_to_fixnum(saving_id)).pluck(:saving_in_year)
```
Comment (phucmh-1356) [inline app/services/year_service.rb]: xóa ()

File: app/services/year_service.rb [hunk 4]
```diff
@@ -30,4 +33,14 @@ def cal_percent status
...
+    saving_num = convert_arr_to_fixnum(saving_in_year)
```
Comment (phucmh-1356) [inline app/services/year_service.rb]:
```suggestion
    convert_arr_to_fixnum saving_in_year
```

File: app/views/shared/_table_intense_report.html.erb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<div class="panel panel-success">
+  <div class="panel-heading">
+    <h1 class="panel-title"><%= t "year_report.title.intents_title"%></h1>
+  </div>
+  <div class="panel-body">
+  <table class="table">
+      <thead class="thead-dark">
+        <tr>
+          <th scope="col"><%= t "year_report.table.name_column.intense"%></th>
```
Comment (phucmh-1356) [inline app/views/shared/_table_intense_report.html.erb]: trước %> cạc 1 spaces ra em

File: app/services/year_service.rb (hunk 1 above)
Comment (hieplt-0914) [reply app/services/year_service.rb]: yes sir

File: app/services/year_service.rb (hunk 3 above)
Comment (hieplt-0914) [reply app/services/year_service.rb]: yes sir

File: app/services/year_service.rb (hunk 2 above)
Comment (hieplt-0914) [reply app/services/year_service.rb]: yes sir

File: app/services/year_service.rb (hunk 4 above)
Comment (hieplt-0914) [reply app/services/year_service.rb]: yes sir

File: app/views/shared/_table_intense_report.html.erb (hunk 5 above)
Comment (hieplt-0914) [reply app/views/shared/_table_intense_report.html.erb]: yes sir

**Comment by hieplt-0914** [comment]
ready

File: app/services/year_service.rb [hunk 6]
```diff
@@ -30,4 +34,14 @@ def cal_percent status
...
   end
+
+  def convert_arr_to_fixnum num
+    num.inject{|n, d| n * 10 + d}
+  end
+
+  def saving_year
+    saving_id = User.by_user_id(@user_id).pluck :saving_id
+    saving_in_year = Saving.by_saving_id(convert_arr_to_fixnum(saving_id)).pluck :saving_in_year
+    saving_num = convert_arr_to_fixnum saving_in_year
```
Comment (phucmh-1356) [inline app/services/year_service.rb]:
```suggestion
    convert_arr_to_fixnum saving_in_year
```

File: app/controllers/report_controller.rb [hunk 7]
```diff
@@ -20,5 +20,7 @@ def year_report
     @expense_amout = yearService.perform_amount_expense
     @savings = standardized(saving(@income_amout, @expense_amout))
     @saving = Saving.new
+
+    @saving_in_year = standardized(yearService.saving_in_year)
```
Comment (phucmh-1356) [inline app/controllers/report_controller.rb]:
```suggestion
    @saving_in_year = standardized yearService.saving_in_year
```

File: app/services/year_service.rb (hunk 6 above)
Comment (hieplt-0914) [reply app/services/year_service.rb]: yes sir

File: app/controllers/report_controller.rb (hunk 7 above)
Comment (hieplt-0914) [reply app/controllers/report_controller.rb]: yes sir

**Comment by hieplt-0914** [comment]
ready

**Comment by phucmh-1356** [comment]
commented
gộp commits đi em

**Comment by hieplt-0914** [comment]
ready

**Pull request merged**